maps-compose/src/main/java/com/google/maps/android/compose/RememberComposeBitmapDescriptor.kt

```diff
@@ -51,6 +51,11 @@ private fun renderComposableToBitmapDescriptor(
         View.MeasureSpec.makeMeasureSpec(parent.height, View.MeasureSpec.AT_MOST),
     )
 
+    if(composeView.measuredWidth == 0 || composeView.measuredHeight == 0) {
+        parent.removeView(composeView)
```

Comment:
You can avoid the extra removeView() by adding try-finally in the right place here.

Reply:
Thanks for the feedback! Yeah, unfortunately I am not sure why this happens in the first place. Maybe some race condition while navigating? I can't dig into it as this happens super rarely and I was not able to reproduce it while having the debugger attached.

I don't think that a different error would help here as we already know that the problem is an incorrect view size. The question is just why we get to this state. As you can see from [the example code](https://github.com/googlemaps/android-maps-compose/issues/518#issue-2105158020) in the issue, the original reporter only uses the GoogleMaps composable and adds some markers (the same as I do in my code). I don't see what we as users of this library could do differently as the code works >99% of the time. Or could we add any meaningful information to a new exception which would help finding the root cause?

Reply:
What if you used `Modifier.requiredSizeIn()` on your marker composable content with a minimum of 1.dp or so, as an alternative workaround? I'd be curious if you still crash then.

Reply:
@ln-12 Actually, I suspect that setting a minimum size on the composable, like I suggested above, would not make a difference.

Here is the problem that I think may be at the root of this issue (have not tested it):
https://github.com/googlemaps/android-maps-compose/blob/0599412f11fb0571c97a52d0b29a4943f2a60a65/maps-compose/src/main/java/com/google/maps/android/compose/RememberComposeBitmapDescriptor.kt#L50-L51

A parent view dimension of 0 would cause this crash, no matter how large the Composable wants to be.

I'd think the parent view here would commonly be the Activity's top ComposeView, or supposedly something else in case of navigation, etc. So the maximum marker icon size would commonly be constrained by the size of the screen, which looks weird to me, as that could be quite large.

A better approach, short of workarounds, may be to use `View.MeasureSpec.UNSPECIFIED` here; restricting marker size by screen size just does not look that useful, it's way too big, creating giant bitmaps. However, I imagine this change would be breaking, as there is no max size anymore. Yet a marker composable relying on something like fillMaxSize is not doing it right, it needs to use something more specific. There may be other concerns with UNSPECIFIED, however, not sure.

Reply:
You are right. I can force the crash by setting the size of the fragment which contains the compose view to 0 (but I am still not sure when this happens while navigating): https://github.com/ln-12/GoogleMapsCrashReproducer/blob/main/app/src/main/res/layout/activity_main.xml#L11-L12

With that, I also tested `View.MeasureSpec.UNSPECIFIED` and for me it did not crash anymore. I can change my PR to use this spec, but just like you, I am not sure about other side effects when doing so.

So what would you suggest going forward?

Reply:
Thank you for these useful observations.

To me `MeasureSpec.UNSPECIFIED` feels like the right thing to do here, as there is no valid relationship between marker size and screen size (assuming marker size is expressed via dips), so relying on parent size for calculating marker size is invalid. `MeasureSpec.AT_MOST` seems to trade more predictable crashes (from potentially relying on absent parent size) for random crashes.

Also, as we see in this case, parent size can change over time, but this would not trigger recalculation of the marker based on the new size.

Not sure if the ComposeView should also set layoutParams, using `LayoutParams.WRAP_CONTENT`. Could some layouts choke on WRAP_CONTENT?

I see that @dkhawk is assigned to this PR, he may need to take over from here.

Reply:
@bubenheimer Thanks for being this to my attention.  I would love talk to you more about it and your many other contributions.  Please reach out to me via my Twitter (@DaleHawkin57064) if that works for you.  Or via LinkedIn: https://www.linkedin.com/in/dalehawkins/.

Reply:
If no one objects, I suggest going with the following (see above comment):
1. Set layoutParams (WRAP_CONTENT) on ComposeView, because docs say you have to.
2. Use View.MeasureSpec.UNSPECIFIED, because this makes the most sense to me. This is a breaking change, as it makes parent size 0. Before, parent size was typically the screen size, which seems an unlikely intentional input for determining marker size and can be obtained in other ways. This change needs to be documented.

Reply:
I changed the PR accordingly. After some testing, I could not get another crash with these changes, so it seems to work.

Reply:
> Should this be documented inline with the code or only as a release note in the changelog?

The requirement for the Composable to have a non-null size in both dimensions needs to be documented in public API KDoc documentation on `MarkerComposable()`. Also in release notes, I believe this would be a task for project maintainers, e.g. @kikoso or @dkhawk.

It could also be flagged prominently in code; personally I'd check for this case after `measure()` and throw an exception of some sort if measure returns 0 for one or both of the dimensions. This way the problem is identified in a well-defined manner if it happens with the ability to give specific instructions on what to do (set a size in the Composable).

Reply:
BTW, I meant "absence of a parent size", not "absence of a parent".

Reply:
@bubenheimer Unfortunately, we now get crashes containing the exact exception from this PR. We use version `6.1.2` of this library. Do you have another idea how this could be solved? Before we forked this lib and implemented an early return which might have led to undesired results but at least did not crash.